Source, rebased on top of 3.0.22-rc2:
https://src.fedoraproject.org/rpms/vlc/c/0eecef99dd1f2ccf9c7059b3f1ca92f04d78c40e?branch=rawhide

From eae5e3786af3157a5c0b78bbb7cee929e670e375 Mon Sep 17 00:00:00 2001
From: Dominik 'Rathann' Mierzejewski <dominik@greysector.net>
Date: Thu, 6 Nov 2025 11:12:17 +0000
Subject: [PATCH 2/2] Fix build with FFmpeg 8

Backport upstream patches to avoid using deprecated fields and
definitions.

Signed-off-by: Andreas Sturmlechner <asturm@gentoo.org>
---
 modules/codec/avcodec/avcommon_compat.h | 47 ++++++++++++++++++++++++-
 1 file changed, 46 insertions(+), 1 deletion(-)

diff --git a/modules/codec/avcodec/avcommon_compat.h b/modules/codec/avcodec/avcommon_compat.h
index 8ab6910f32..b476c54004 100644
--- a/modules/codec/avcodec/avcommon_compat.h
+++ b/modules/codec/avcodec/avcommon_compat.h
@@ -75,9 +75,12 @@
 #ifndef AV_CODEC_CAP_SMALL_LAST_FRAME
 # define AV_CODEC_CAP_SMALL_LAST_FRAME CODEC_CAP_SMALL_LAST_FRAME
 #endif
-#ifndef AV_INPUT_BUFFER_MIN_SIZE
+#if !defined(AV_INPUT_BUFFER_MIN_SIZE) && defined(FF_MIN_BUFFER_SIZE)
 # define AV_INPUT_BUFFER_MIN_SIZE FF_MIN_BUFFER_SIZE
 #endif
+#ifndef AV_INPUT_BUFFER_MIN_SIZE
+# define AV_INPUT_BUFFER_MIN_SIZE 16384
+#endif
 #ifndef  FF_MAX_B_FRAMES
 # define  FF_MAX_B_FRAMES 16 // FIXME: remove this
 #endif
@@ -87,6 +90,48 @@
 #ifndef FF_API_STRUCT_VAAPI_CONTEXT
 # define FF_API_STRUCT_VAAPI_CONTEXT (LIBAVCODEC_VERSION_MAJOR < 59)
 #endif
+#ifndef FF_PROFILE_UNKNOWN // removed in ffmpeg 8
+# define FF_PROFILE_UNKNOWN AV_PROFILE_UNKNOWN
+#endif
+#ifndef FF_LEVEL_UNKNOWN
+# define FF_LEVEL_UNKNOWN AV_LEVEL_UNKNOWN
+#endif
+#ifndef FF_PROFILE_HEVC_MAIN
+# define FF_PROFILE_HEVC_MAIN AV_PROFILE_HEVC_MAIN
+#endif
+#ifndef FF_PROFILE_HEVC_MAIN_10
+# define FF_PROFILE_HEVC_MAIN_10 AV_PROFILE_HEVC_MAIN_10
+#endif
+#ifndef FF_PROFILE_VP9_0
+# define FF_PROFILE_VP9_0 AV_PROFILE_VP9_0
+#endif
+#ifndef FF_PROFILE_VP9_2
+# define FF_PROFILE_VP9_2 AV_PROFILE_VP9_2
+#endif
+#ifndef FF_PROFILE_AAC_LOW
+# define FF_PROFILE_AAC_LOW AV_PROFILE_AAC_LOW
+#endif
+#ifndef FF_PROFILE_AAC_MAIN
+# define FF_PROFILE_AAC_MAIN AV_PROFILE_AAC_MAIN
+#endif
+#ifndef FF_PROFILE_AAC_SSR
+# define FF_PROFILE_AAC_SSR AV_PROFILE_AAC_SSR
+#endif
+#ifndef FF_PROFILE_AAC_LTP
+# define FF_PROFILE_AAC_LTP AV_PROFILE_AAC_LTP
+#endif
+#ifndef FF_PROFILE_AAC_HE_V2
+# define FF_PROFILE_AAC_HE_V2 AV_PROFILE_AAC_HE_V2
+#endif
+#ifndef FF_PROFILE_AAC_HE
+# define FF_PROFILE_AAC_HE AV_PROFILE_AAC_HE
+#endif
+#ifndef FF_PROFILE_AAC_LD
+# define FF_PROFILE_AAC_LD AV_PROFILE_AAC_LD
+#endif
+#ifndef FF_PROFILE_AAC_ELD
+# define FF_PROFILE_AAC_ELD AV_PROFILE_AAC_ELD
+#endif
 
 #endif /* HAVE_LIBAVCODEC_AVCODEC_H */
 
-- 
2.52.0
